### PR TITLE
misc fixes

### PR DESCRIPTION
**Pull request by cmyui**
felt like doing some code review to see how u guys have been progressing - here some stuff i found while going thru it

### CODE REVIEW

File: conn/mysql.py [hunk 1]
```diff
@@ -164,5 +163,5 @@ def kill(self) -> None:
                 of this coro.
         """
 
-        self._pool.terminate()
         self._pool.close()
+        await self._pool.wait_closed()
```
Comment (cmyui) [inline conn/mysql.py:167]: matches underlying library protocol

**Review by cmyui — COMMENTED** [review]
if you're using vscode, one thing i'd recommend is turning up the validation level of your type checker - this code has lots of typing errors and i think just having this visual queue while developing would probably help a lot

image

File: constants/c_modes.py [hunk 2]
```diff
@@ -10,11 +10,11 @@ class CustomModes(IntEnum):
     RELAX = 1
     AUTOPILOT = 2
 
-    def to_db_suffix(self):
+    def to_db_suffix(self) -> str:
         """Returns the database table (for redis and sql) suffix for the given
         `c_mode`."""
 
-        return _db_suffixes[self.value]
+        return _db_suffixes[self]
```
[the remark this answers is not in the document]
Comment (cmyui) [reply constants/c_modes.py:17]: there are a few other instances of this in the PR

File: objects/beatmap.py [hunk 3]
```diff
@@ -339,7 +339,7 @@ async def try_update(self) -> Optional["Beatmap"]:
         self.status = Status.UPDATE_AVAILABLE      
         
         # Delete cached `.osu`
-        delete_osu_file(self.id)
+        await delete_osu_file(self.id)
```
Comment (cmyui) [inline objects/beatmap.py:342]: :flushed:

**Comment by RealistikDash** [comment]
Hey, cheers for the PR!

At quick glance, your PR looks well-made, making important and valid changes to the USSR codebase alongside rightfully displaying areas which deserve the most improvement and how to proceed. While the changes all look good, merging this PR requires testing on our end which should be done by tomorrow once our USSR dev environment is set up.

**Pull request merged**